Infrastructure/SolutionRepository.cs

```diff
@@ -69,16 +69,16 @@ public SolutionRepository(IDbConnectionFactory dbConnection, ILogger<SolutionRep
 		}
 	}
 
-	public async Task<bool> CheckAnonUserExistsInSessionAsync(int userId)
+	public async Task<bool> CheckAnonUserExistsInSessionAsync(int userId, int sessionId)
 	{
 		using var con = await _dbConnection.CreateConnectionAsync();
 		var query = """
 		            SELECT COUNT(*) FROM exercise_in_session AS eis
 		                JOIN anon_users AS au
 		                ON au.session_id = eis.session_id
-		            WHERE au.user_id = @UserId;
+		            WHERE au.user_id = @UserId AND eis.session_id = @SessionId;
```

Comment:
The query could maybe be simplified since the calling site for this function is SubmitSolutionAsync.
Where the session id is already provided in the dto, such that a query could either look like this:
```sql
SELECT session_id FROM anon_users WHERE user_id = x
```
and then match on the session_id retrieved from the incoming dto
 or 
 ```sql
SELECT Count(*) FROM anon_users WHERE user_id = x AND session_id = x
 ```
make sure the count is above 0. 

Since we are essentially doing just that I think we can avoid jumping some join hoops.